sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/BeginInvokePairedWithEndInvoke.cs

```diff
@@ -268,9 +267,15 @@ IAsyncResult BeginInvokeHiddenInALocalFunction()
                     return caller.BeginInvoke("method", null, null); // Noncompliant
                 }
             }
-
         }
-
     }
 
+    internal class AsyncResult : IAsyncResult
```

Comment:
why is this necessary?

Reply:
In order to be able to keep the tests compatible with both .Net Framework and .Net Core. `System.Runtime.Remoting.Messaging` was used just for adding an implementation for **IAsyncResult** but the implementation is not available on .Net Core so I made a dummy one and removed the namespace.

Reply:
Sounds good, please add a comment